src/components/post-card/PostCard.js

```diff
@@ -8,57 +8,83 @@ import './PostCard.scss';
 //  в ней должно появиться свойство state
 //  в котором будет флажок showComments
 //  и будет кнопка, лейба которой будет либо "show comments" либо "hide comments"
+const buttonLabel = ['Show COMMENTS','Hide COMMENTS']
 
-export function PostCard(props) {
-  const { post, hasImage, comments = [], author } = props;
+export class PostCard extends Component {
+  constructor( props ) {
+    super( props );
 
-  console.log('hasImage', hasImage);
-  const { title, body } = post;
+    this.state = {
+      showComments: false,
+      selectedLabel: buttonLabel[0]
+    }
+  }
 
-  const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+  changeButton = () => {
+    const  [ show, hide ] = buttonLabel;
 
-  const renderImage = () => {
-    return hasImage ? (
-      <img src={kittyUrl} />
-    ) : (
-      <img src={DefaultImg} />
-    );
+    this.setState({showComments:!this.state.showComments});
+    this.state.showComments === false ?
+      this.setState({selectedLabel:hide}) :
+      this.setState({selectedLabel:show})
```

Comment:
This code wouldn't be needed

Reply:
This code will not working properly all the time
`setState ` is async function, and when the execution would come to line 27 the code from line 26 will not be already executed

in this case you can use setState only once like this:
```
this.setState({
 showComments:!this.state.showComments,
 selectedLabe: !this.state.showComments ? hide : show
});
```